fdbclient/BlobGranuleFiles.cpp

```diff
@@ -220,7 +220,8 @@ void validateEncryptionHeaderDetails(const BlobGranuleFileEncryptionKeys& eKeys,
                                      const BlobCipherEncryptHeader& header,
                                      const StringRef& ivRef) {
 	// Validate encryption header 'cipherHeader' details sanity
-	if (!(header.cipherHeaderDetails.baseCipherId == eKeys.headerCipherKey->getBaseCipherId() &&
+	if (header.cipherHeaderDetails.isValid() &&
```

Comment:
a nit: BlobCipherDetails overrides  `==` and `!=` operators, we can simplify the code by just comparing the objects instead of individual elements.


	```
bool operator==(const BlobCipherDetails& o) const {
		return encryptDomainId == o.encryptDomainId && baseCipherId == o.baseCipherId && salt == o.salt;
	}
	bool operator!=(const BlobCipherDetails& o) const { return !(*this == o); }
```